calico/openstack/mech_calico.py

```diff
@@ -668,11 +678,14 @@ def resync_endpoints(self, context):
             for port in missing_ports:
                 # Fill out other information we need on the port and write to
                 # etcd.
-                self.add_port_gateways(port, context)
-                self.add_port_interface_name(port)
+                port['fixed_ips'] = self.get_fixed_ips_for_port(
+                    context, port
+                )
                 port['security_groups'] = self.get_security_groups_for_port(
                     context, port
                 )
+                self.add_port_gateways(port, context)
+                self.add_port_interface_name(port)
                 self.transport.endpoint_created(port)
```

Comment:
We've made the same change in 4 very similar blocks of code here.  Surely time for a subroutine!?